### PR TITLE
Face segmentation

### PR DESCRIPTION
**Pull request by kvrooman**
WIP for code pipeline to create facial segmentation masks for a directory of images

Pre-trained model for face segmentation
"http://drive.google.com/uc?export=view&id=1eaAG9sYsxes4NPekZ2Q5be9AmqYDeYQP"

Before additional post-processing
test 3

test 27

test 16

test 39

### CODE REVIEW

**Comment by iperov** [comment]
what is faceseg ? what segs it generates?
is it work on side faces?

**Comment by kvrooman** [comment]
@iperov A Keras implementation of https://github.com/YuvalNirkin/face_segmentation by converting from Caffe and importing their pre-trained weights

**Comment by iperov** [comment]
it doesnt work with side faces

**Comment by ruah1984** [comment]
@kvrooman , i found you have merge the face segmentation into the train_refactor branch.
may i know how we can activate the face segmentation during the training in train_refactor branch ?? will it automatic start when we do the training??

**Comment by kvrooman** [comment]
> it doesnt work with side faces

**Comment by iperov** [comment]
@kvrooman why it produces so imprecise mask?

I trained fanseg that produces smooth mask.

python_2019-04-19_21-57-12

**Comment by iperov** [comment]
if you want, you can try to train your own seg model
my dataset located in
https://mega.nz/#F!b9MzCK4B!zEAG9txu7uaRUjXz9PtBqg
contains 3277 manually refined and segmented FAN-aligned faces. 
Fixed mask produced by include/exclude polygons ( IEPolys.py class )

**Comment by torzdf** [comment]
@iperov Thanks! 👍 We'll take a look

**Comment by iperov** [comment]
NSFW example using fanseg https://mrdeepfakes.com/video/3254/angelina-fan-dst-sample

**Comment by kvrooman** [comment]
> @kvrooman why it produces so imprecise mask?
> 
> I trained fanseg that produces smooth mask.
> 

I haven't been able to work on this recently, but thought I'd point out that the network produces smooths masks natively just like your example above. I purposely binarize them in the post-processing at a certain threshold.

Looks like you've been manually annotating training examples with your mask editor to feed into your new training.  I know that takes a good amount of time so I appreciate the effort and congratulate you. 

I can run my training examples through this new network to get an idea how well it generalizes. I was shown a previous mask training model from your github repo as of about a week or two ago so it looks like you've made some good progress with the new one.  

"why it produces so imprecise mask". Lol, I always love your friendly competitive banter!
